currency/views.py

```diff
@@ -19,10 +19,27 @@ def create_currency(request):
             currency.save()
             return redirect('/currencies')
     context = {'form': form}
-    return render(request, 'currency/create_currency.html', context)
+    return render(request, 'currency/currency_form.html', context)
 
 
-def update_currency(request):
-    form = CurrencyForm()
+def update_currency(request, pk):
+    currency = Currency.objects.get(id=pk)
+    form = CurrencyForm(instance=currency)
+    if request.method == 'POST':
+        form = CurrencyForm(request.POST, request.FILES, instance=currency)
+        if form.is_valid():
+            currency = form.save(commit=False)
+            currency.user = request.user
+            currency.save()
+            return redirect('/currencies')
     context = {'form': form}
-    return render(request, 'currency/create_currency.html', context)
+    return render(request, 'currency/currency_form.html', context)
+
+
+def delete_currency(request, pk):
+    currency = Currency.objects.get(id=pk)
+    if request.method == 'POST':
```

Comment:
Why `DELETE` method is not used?

Reply:
I've just followed the tutorial but after your question I've searched a bit and found that allowed methods in forms only GET and POST. [https://softwareengineering.stackexchange.com/questions/114156/why-are-there-are-no-put-and-delete-methods-on-html-forms](url)